sensors/repository_sensor.py

```diff
@@ -0,0 +1,186 @@
+            ))
+            self._init_cloud_last_commit()
+        else:
+            raise ValueError('specified bitbucket type (%s) is not supported' % self.service_type)
+        self._increment_event_id()
+
+    def poll(self):
+        new_commits = []
+        if self.service_type == 'server':
+            new_commits = self._get_server_updated_commits()
+        elif self.service_type == 'cloud':
+            new_commits = self._get_cloud_updated_commits()
+
+        if new_commits:
+            self._dispatch_trigger_for_server('commit', new_commits)
+
+    def cleanup(self):
+        pass
+
+    def add_trigger(self, trigger):
+        pass
+
+    def update_trigger(self, trigger):
+        pass
+
+    def remove_trigger(self, trigger):
+        pass
+
+    def _poll_bitbucket_server(self):
+        pass
+
+    def _poll_bitbucket_cloud(self):
+        pass
+
+    def _dispatch_trigger_for_server(self, event_type, payload):
+        data = {
+            'id': self._get_event_id(),
+            'created_at': datetime.now().strftime(self.DATE_FORMAT),
+            'type': event_type,
+            'payload': payload,
+        }
+        self._increment_event_id()
+
+        self._sensor_service.dispatch(trigger=self._trigger_ref, payload=data)
+
+    # Returns unique id for the dispatching events
+    def _get_event_id(self):
+        ret_id = self._sensor_service.get_value(name=self._trigger_ref)
+        if not ret_id:
+            return 1
+        return ret_id
+
+    # Increments event id of datastore
+    def _increment_event_id(self):
+        self._sensor_service.set_value(name=self._trigger_ref,
+                                       value=int(self._get_event_id()) + 1)
+
+    # initialize last commit for each branches
+    def _init_server_last_commit(self):
+        [[self._set_last_commit('%s/%s' % (p, r), b['displayId'], b['latestCommit'])
+            for b in self.client.projects[p].repos[r].branches()]
+            for (p, r) in [x.split('/') for x in self.repositories]]
+
+    def _init_cloud_last_commit(self):
+        [[self._set_last_commit('%s/%s' % (o, r), b.name, b.commits().next().hash)
+            for b in Branch.find_branches_in_repository(repository_name=r, client=self.client,
+                                                        owner=o) if isinstance(b, Branch)]
+            for (o, r) in [x.split('/') for x in self.repositories]]
+
+    def _set_last_commit(self, repo, branch, last_commit):
+        if repo not in self.last_commit:
+            self.last_commit[repo] = {}
+        self.last_commit[repo][branch] = last_commit
+
+    def _get_server_updated_commits(self):
+        new_commits = []
+        for (proj, repo) in [x.split('/') for x in self.repositories]:
+            repo_name = '%s/%s' % (proj, repo)
+            for branch in self.client.projects[proj].repos[repo].branches():
+                for commit in self.client.projects[proj].repos[repo].commits(branch['id']):
```

Comment:
How does this work if there are many commits?

Does `commits()` return some kind of generator or it only returns up to X last commits?

In any case, if possible, it would be great if we could limit number of recent commits returned here or similar (e.g. only return new commits since TIMESTAMP).

Otherwise it could get quite expensive and slow constantly polling and iterating over 1000s of commits during each poll interval.